Gems/Atom/Feature/Common/Assets/Shaders/SkyAtmosphere/SkyAtmosphereCommon.azsli

```diff
@@ -29,11 +29,11 @@ option bool o_enableAerialPerspective = true;
 
 ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
 {
-    Texture2DMS<float> m_depth;
+    Texture2D<float2> m_depth;
```

Comment:
Actually Use the supervariant system to switch between msaa and non-msaa. So you can leave this as Texture2DMS. Add these params to the .shader file using this srg

```
"Supervariants":
    [
        {
            "Name": "NoMSAA",
            "AddBuildArguments": {
                "azslc": ["--no-ms"]
            }
        }
    ]
```
This will ensure that the shader pipeline will create two variants, one with Texture2DMS and one with Texture2D. Then at runtime RPI should have the current supervariant name set based on msaa sample count (SetSupervariantName) from the pipeline used. So when the shader is loaded it should load the correct one based on the supervariant name. If the supervariant name is "" then it will use  Texture2DMS but if it is set to "nomsaa" it will use Texture2D. This is how we switch between the two for everywhere else in the engine.

Reply:
you are my hero, I was going crazy trying to figure out how to do this in a simple way 😄